spec/passfort/http_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Passfort::Http do
+  shared_examples "an error response" do
+    before do
+      stub_request(method, api_path).
+        to_return(status: status, body: body)
+    end
+
+    # rubocop:disable RSpec/MultipleExpectations
+    it "raises a specific APIError" do
+      is_expected.to raise_error do |error|
+        expect(error).to be_a(error_class)
+      end
+    end
+    # rubocop:enable RSpec/MultipleExpectations
+  end
+
+  shared_examples "an API call that handles all errors" do
+    context "when returning a request error" do
+      let(:status) { 404 }
+      let(:body) { "{}" }
+      let(:error_class) { Passfort::Errors::RequestError }
+
+      it_behaves_like "an error response"
```

Comment:
Thought I'd point out that you can write all these tests without a shared example in the same or fewer lines of code
```ruby
before { stub_request(method, api_path).to_return(status: status, body: body) }

context "when returning a request error" do
  let(:status) { 404 }
  let(:body) { "{}" }
  let(:error_class) { Passfort::Errors::RequestError }
  
  it { is_expected.to raise_error(instance_of(error_class)) }
end

context "when returning an invalid API Key error" do
  ...
end
```